lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm

```diff
@@ -20,57 +20,71 @@ use Mojo::File;
 use POSIX 'strftime';
 
 sub index {
-    my $self   = shift;
+    my ($self, $obs_project, $folders) = @_;
     my $helper = $self->obs_rsync;
     my %folder_info_by_name;
-    my $folders = Mojo::File->new($helper->home)->list({dir => 1})->grep(sub { -d $_ })->map('basename')
-      ->grep(qr/^^(?!t$|profiles$|script$|xml$)/)->to_array;
+    if (!$folders) {
+        $folders
+          = Mojo::File->new($helper->home, $obs_project // "")->list({dir => 1})->grep(sub { -d $_ })->map('basename')
+          ->to_array;
+        @$folders = grep { !/^(t|profiles|script|xml)$/ } @$folders;
+    }
 
     for my $folder (@$folders) {
-        my $run_last_info = $helper->get_run_last_info($folder);
-        my ($fail_last_job_id, $fail_last_when) = $helper->get_fail_last_info($folder);
+        my $batch = $folder;
+        $batch = "$obs_project|$folder" if $obs_project;
+        my $run_last_info = $helper->get_run_last_info($batch);
+        my ($fail_last_job_id, $fail_last_when) = $helper->get_fail_last_info($batch);
         $folder_info_by_name{$folder} = {
             run_last         => $run_last_info->{dt},
             run_last_version => $run_last_info->{version},
             run_last_job_id  => $run_last_info->{job_id},
             fail_last_when   => $fail_last_when,
             fail_last_job_id => $fail_last_job_id,
-            dirty_status     => $helper->get_dirty_status($folder)};
+            dirty_status     => $helper->get_dirty_status($obs_project // $folder)};
     }
 
-    my $running_jobs = $self->app->minion->backend->list_jobs(0, undef,
-        {tasks => ['obs_rsync_run'], states => ['active', 'inactive']});
-
-    for my $job (@{$running_jobs->{jobs}}) {
-        my $args = $job->{args};
-        $args = $args->[0] if (ref $args eq 'ARRAY' && scalar(@$args) == 1);
-
-        next unless (ref $args eq 'HASH' && scalar(%$args) == 1 && $args->{project});
-        my $project = $args->{project};
-        next unless exists $folder_info_by_name{$project};
-        $folder_info_by_name{$project}->{state} = $job->{state};
-        my $created_at = $job->{created};
-        if ($created_at) {
-            $created_at = strftime('%Y-%m-%d %H:%M:%S %z', localtime($created_at));
-            $folder_info_by_name{$project}->{created} = $created_at;
+    if (!$obs_project) {
+        my $running_jobs = $self->app->minion->backend->list_jobs(0, undef,
+            {tasks => ['obs_rsync_run'], states => ['active', 'inactive']});
```

Comment:
The second argument for `list_jobs` is not supposed to be `undef`. That might just break in a future Minion update. It's safer to pick a very large number.